src/leaderboard.js

```diff
@@ -1,24 +1,64 @@
-const leaderboardData = [
-  { name: 'Player 1', score: 100 },
-  { name: 'Player 2', score: 20 },
-  { name: 'Player 3', score: 50 },
-  { name: 'Player 4', score: 78 },
-  { name: 'Player 5', score: 125 },
-  { name: 'Player 6', score: 77 },
-  { name: 'Player 7', score: 42 },
-];
-
-function renderLeaderboard() {
+const leaderboardData = [];
+
+async function renderLeaderboard() {
   const leaderboardTable = document.getElementById('leaderboard').getElementsByTagName('tbody')[0];
 
+  // Clear the existing table rows
+  while (leaderboardTable.firstChild) {
+    leaderboardTable.removeChild(leaderboardTable.firstChild);
+  }
+
+  try {
+    // Fetch data from the API
+    const response = await fetch('https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/Zl4y7IVkemOTTVg2fUdz/scores/');
+    if (!response.ok) {
+      throw new Error('Failed to fetch data from the API');
+    }
+
+    const data = await response.json();
+
+    // Update leaderboardData with API data
+    leaderboardData.length = 0;
+    leaderboardData.push(...data.result);
+  } catch (error) {
+    console.error('Error fetching data from the API:', error);
+    alert('Failed to fetch data from the API. Please try again.');
+  }
+
+  // Render the updated data
   leaderboardData.forEach((player) => {
     const row = leaderboardTable.insertRow();
     const nameCell = row.insertCell(0);
     const scoreCell = row.insertCell(1);
 
-    nameCell.textContent = player.name;
+    nameCell.textContent = player.user;
     scoreCell.textContent = player.score;
   });
 }
 
-export default renderLeaderboard;
+async function addScoreToAPI(name, score) {
```

Comment:
- [ ] Please kindly make use of `arrow functions` rather than using the function key word to create your `renderLeaderboard and addToAPI` functionality 👍 , it is important to follow and use the new `ES 6` features in your application